test/suites/integration/access_log_policy_test.go

```diff
@@ -81,45 +82,77 @@ var _ = Describe("Access Log Policy", Ordered, func() {
 		logGroup2Arn      string
 		deliveryStreamArn string
 		roleArn           string
-		bucketName        = "eks-gateway-api-access-log-test-" + strconv.Itoa(rand.Int())
+		awsResourceName   string
+		sess              = session.Must(session.NewSession(&aws.Config{Region: aws.String(config.Region)}))
 	)
 
 	BeforeAll(func() {
+		awsResourceName = awsResourceNamePrefix + utils.RandomAlphaString(10)
+
+		tags := testFramework.NewTestTags(testSuite)
+
 		// Create S3 Bucket
-		s3Client = s3.New(session.Must(session.NewSession(&aws.Config{Region: aws.String(config.Region)})))
+		s3Client = s3.New(sess)
 		_, err := s3Client.CreateBucketWithContext(ctx, &s3.CreateBucketInput{
-			Bucket: aws.String(bucketName),
+			Bucket: aws.String(awsResourceName),
 		})
 		Expect(err).To(BeNil())
-		bucketArn = "arn:aws:s3:::" + bucketName
+		bucketArn = "arn:aws:s3:::" + awsResourceName
+
+		// Tag S3 Bucket
+		tagging := &s3.Tagging{
+			TagSet: make([]*s3.Tag, 0),
+		}
+		for key, value := range tags {
+			tagging.TagSet = append(tagging.TagSet, &s3.Tag{
+				Key:   aws.String(key),
+				Value: value,
+			})
```

Comment:
aws sdk is annoying with it's duplicated Tag types...

Reply:
A universal tag type would be great